IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -217,4 +217,103 @@ For asymmetric keys this holds the public key hash which allows digital identity
+        {
+            return legacyKeys;
+        }
+    }
+    
+    if (!tenantId)
+    {
+        // default registration should have a tenantId associated.
+        return legacyKeys;
+    }
+    
+    NSString *tag = nil;
+    __unused MSIDWPJKeyPairWithCert *defaultKeys = nil;
+    NSString *defaultSharedAccessGroup = [NSString stringWithFormat:@"%@.com.microsoft.workplacejoin.v2", teamId];
+    tag = [NSString stringWithFormat:@"%@#%@", kWPJPrivateKeyIdentifier, tenantId];
+    tagData = [tag dataUsingEncoding:NSUTF8StringEncoding];
+    // Default registrations can be done using RSA/ECC in iOS and only ECC in macOS. 1st Looking for RSA device key in the keychain.
+    __unused NSDictionary *extraDefaultPrivateKeyAttributes = @{ (__bridge id)kSecAttrApplicationTag : tagData,
+                                                        (__bridge id)kSecAttrAccessGroup : defaultSharedAccessGroup,
+                                                        (__bridge id)kSecAttrKeyType : (__bridge id)kSecAttrKeyTypeRSA };
+    
+    extraCertAttributes = @{ (__bridge id)kSecAttrAccessGroup : defaultSharedAccessGroup };
+    // In macOS, default registrations can only be ECC. Skip checking default RSA registration for macOS.
+#if !TARGET_OS_OSX
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"Checking keychain for default registration done using RSA key.");
+    defaultKeys = [self findWPJRegistrationInfoWithAdditionalPrivateKeyAttributes:extraDefaultPrivateKeyAttributes certAttributes:extraCertAttributes context:context];
+
+    // If secondary Identity was found, return it
+    if (defaultKeys)
+    {
+        defaultKeys.keyChainVersion = MSIDWPJKeychainAccessGroupV2;
+        return defaultKeys;
+    }
+#endif
+    
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"Checking keychain for default registration done using ECC key.");
+    // Since the defualt RSA search returned nil in iOS, the key might be an ECC key. Use the tag specific for EC device key and re-try
+    tag = [NSString stringWithFormat:@"%@%@", tag, kECPrivateKeyTagSuffix];
+    tagData = [tag dataUsingEncoding:NSUTF8StringEncoding];
+    NSMutableDictionary *privateKeyAttributes = [[NSMutableDictionary alloc] initWithDictionary:extraDefaultPrivateKeyAttributes];
+    [privateKeyAttributes setObject:tagData forKey:(__bridge id)kSecAttrApplicationTag];
+    // Not including kSecAttrTokenIDSecureEnclave in query dict as in the future registrations maybe ECC based even in software keychain
+    [privateKeyAttributes setObject:(__bridge id)kSecAttrKeyTypeECSECPrimeRandom forKey:(__bridge id)kSecAttrKeyType];
+    [privateKeyAttributes setObject:@256 forKey:(__bridge id)kSecAttrKeySizeInBits];
+    defaultKeys = [self findWPJRegistrationInfoWithAdditionalPrivateKeyAttributes:privateKeyAttributes certAttributes:extraCertAttributes context:context];
+    if (defaultKeys)
+    {
+        defaultKeys.keyChainVersion = MSIDWPJKeychainAccessGroupV2;
```

Comment:
Add logs regarding where keys are returned from

Reply:
Add info level logs wherever private key is returned